### PR TITLE
Switch from log to klog

### PR DESCRIPTION
**Pull request by michaelkolber**
ref: #18726

Also:
- Gracefully handlesome previously fatal errors
- Make some fatal errors fail fast

/cc BenTheElder spiffxp

### CODE REVIEW

File: triage/update_summaries.sh [hunk 1]
```diff
@@ -72,7 +72,9 @@ mkdir -p slices
   --builds triage_builds.json \
   --tests triage_tests/*.json \
   --output failure_data.json \
-  --output_slices slices/failure_data_PREFIX.json
+  --output_slices slices/failure_data_PREFIX.json \
+  --logtostderr=true \
```
Comment (michaelkolber) [inline triage/update_summaries.sh:76]: I'm pretty sure this is wrong but I'm not sure how else to pass in the flag

Comment (BenTheElder) [reply triage/update_summaries.sh:76]: binaries frequently set this themselves / hide the flag

Comment (BenTheElder) [reply triage/update_summaries.sh:76]: https://cs.k8s.io/?q=logtostderr%22&i=nope&files=&repos=

Comment (michaelkolber) [reply triage/update_summaries.sh]: Setting it internally does not make the flag work

Comment (michaelkolber) [reply triage/update_summaries.sh]: I switched to klog/v2 _and_ called InitFlags, that should do it

Comment (michaelkolber) [reply triage/update_summaries.sh]: Added a custom flagset as per https://github.com/kubernetes/klog/blob/master/examples/coexist_glog/coexist_glog.go

Comment (michaelkolber) [reply triage/update_summaries.sh]: Ok, it seems to have passed without any issues

**Comment by michaelkolber** [comment]
This is ready for review

File: triage/summarize/cluster.go [hunk 2]
```diff
@@ -68,7 +70,7 @@ func clusterLocal(failuresByTest failuresGroup, numWorkers int) nestedFailuresGr
 
 	numFailures := 0 // The number of failures processed so far
 	start := time.Now()
-	logInfo("Clustering failures for %d unique tests...", len(failuresByTest))
+	klog.V(2).Infof("Clustering failures for %d unique tests...", len(failuresByTest))
```
Comment (MushuEE) [inline triage/summarize/cluster.go:73]: What is enabling the V-level logging? Will it log nothing if not enabled? https://godoc.org/github.com/kubernetes/klog#Verbose.Enabled

Comment (michaelkolber) [reply triage/summarize/cluster.go:73]: I think by default it logs at V2, but regardless it's being set here:  
https://github.com/kubernetes/test-infra/pull/18858/files#diff-e209704c6b63bec066319227a3d98099R94

**Pull request merged**